ssh_automated_tests/core/shell.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from time import sleep
+
+from ssh_automated_tests.core.connection import Connection
+
+
+class Shell(ABC):
+    @abstractmethod
+    def execute(self, command: str):
+        pass
+
+    @abstractmethod
+    def execute_with_output(self, command: str):
+        pass
+
+    @abstractmethod
+    def close(self):
+        pass
+
+
+class InteractiveShell(Shell):
```

Comment:
I suggest this commit can be split to three commits as it affects different functionalities.
For example:
1) Changes done in connection.py
2) Adding channel.py
3) Adding shell.py and removal of cmd_handler